page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/utils/ExpectedConditionsUtils.java

```diff
@@ -123,101 +124,60 @@ public static void waitUntilElementGoneFromDom(WebElement webElement, long timeo
             }
-            sleep(1);
+            sleep(PROPERTIES.getTimeout());
         }
         throw new NoSuchElementException("Timed out after " + timeout + " milliseconds waiting for web element '" + webElement.toString() + "' gone from DOM");
     }
 
     /**
-     * @param element a {@link org.openqa.selenium.WebElement} object.
+     * @param element a {@link WebElement} object.
      */
     public static void waitUntilElementGetInvisible(WebElement element) {
-        new WebDriverWait(PageFactory.getDriver(), PageFactory.getTimeOutInSeconds())
+        new WebDriverWait(PageContext.getCurrentPage().getDriver(), PROPERTIES.getTimeout())
                 .until(ExpectedConditions.not(ExpectedConditions.visibilityOf(element)));
     }
 
     /**
      * <p>
      * waitForElementGetEnabled.</p>
      *
-     * @param webElement a {@link org.openqa.selenium.WebElement} object.
-     * @param timeout    a long.
-     * @throws ru.sbtqa.tag.pagefactory.exceptions.WaitException TODO
+     * @param webElement a {@link WebElement} object.
+     * @param timeout in milliseconds
+     * @throws WaitException TODO
      */
     public static void waitForElementGetEnabled(WebElement webElement, long timeout) throws WaitException {
         long timeoutTime = DateManager.getCurrentTimestamp() + timeout;
         while (timeoutTime > DateManager.getCurrentTimestamp()) {
-            sleep(1);
             try {
                 if (webElement.isEnabled()) {
                     return;
                 }
             } catch (Exception e) {
                 LOG.debug("Target element still not enable", e);
             }
-
+            sleep(1);
         }
         throw new WaitException("Timed out after '" + timeout + "' milliseconds waiting for availability of '" + webElement + "'");
     }
 
     /**
-     * Accept any alert regardless of its message
-     *
-     * @throws ru.sbtqa.tag.pagefactory.exceptions.WaitException if alert didn't appear during timeout
-     */
-    public static void acceptAlert() throws WaitException {
-        interactWithAlert("", true);
-    }
-
-    /**
-     * Dismiss any alert regardless of its message
-     *
-     * @throws ru.sbtqa.tag.pagefactory.exceptions.WaitException if alert didn't appear during timeout
-     */
-    public static void dismissAlert() throws WaitException {
-        interactWithAlert("", false);
-    }
-
-    /**
-     * Wait for an alert with corresponding text (if specified). Depending on the decision, either accept it or decline
-     * If messageText is empty, text doesn't matter
+     * <p>
+     * waitForElementGetEnabled.</p>
      *
-     * @param messageText text of an alert. If empty string is provided, it is being ignored
-     * @param decision    true - accept, false - dismiss
-     * @throws WaitException in case if alert didn't appear during default wait timeout
+     * @param webElement a {@link WebElement} object.
+     * @throws WaitException TODO
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)